package pyqtc.pb;

import "codemodel.proto";

message Message {
  optional int32 id = 1;
  
optional ErrorResponse error_response = 2;

  optional ParseFileRequest parse_file_request = 3;
  optional ParseFileResponse parse_file_response = 4;
  
  optional GetPythonPathRequest get_python_path_request = 5;
  optional GetPythonPathResponse get_python_path_response = 6;
}

message ParseFileRequest {
  optional string filename = 1;
}

message ParseFileResponse {
  message SyntaxError {
    optional Position position = 1;
    optional string text = 2;
  }

  optional Scope module = 1;
  optional SyntaxError syntax_error = 2;
}

message GetPythonPathRequest {
}

message GetPythonPathResponse {
  repeated string path_entry = 1;
}

message ErrorResponse {
  optional string message = 1;
}
